### ISSUE TITLE
new NetworkManager 1.41 breaks azure

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#1341]
Our Azure tests have been failing with what looked like an infrastructure flake but it turns out the instances weren't coming up because of a change in NetworkManager in the disk image. I tracked back to when the failures started happening and it was the transition from `38.20221102.91.0`->`38.20221103.91.0` where:

```
Upgraded:

    NetworkManager 1:1.40.0-1.fc38.x86_64 → 1:1.41.4-2.fc38.x86_64
    NetworkManager-cloud-setup 1:1.40.0-1.fc38.x86_64 → 1:1.41.4-2.fc38.x86_64
    NetworkManager-libnm 1:1.40.0-1.fc38.x86_64 → 1:1.41.4-2.fc38.x86_64
    NetworkManager-team 1:1.40.0-1.fc38.x86_64 → 1:1.41.4-2.fc38.x86_64
    NetworkManager-tui 1:1.40.0-1.fc38.x86_64 → 1:1.41.4-2.fc38.x86_64
    elfutils-default-yama-scope 0.187-8.fc38.noarch → 0.188-2.fc38.noarch
    elfutils-libelf 0.187-8.fc38.x86_64 → 0.188-2.fc38.x86_64
    elfutils-libs 0.187-8.fc38.x86_64 → 0.188-2.fc38.x86_64
    glib2 2.74.1-1.fc38.x86_64 → 2.74.1-2.fc38.x86_64
    json-c 0.16-2.fc37.x86_64 → 0.16-3.fc38.x86_64
    rpm-ostree 2022.15-2.fc38.x86_64 → 2022.15-3.fc38.x86_64
    rpm-ostree-libs 2022.15-2.fc38.x86_64 → 2022.15-3.fc38.x86_64 
```

I did a test with NM pinned to `NetworkManager-1.40.0-1.fc38` and the instances come up again. Unfortunately I'm not yet sure what change in NetworkManager it is. For the most part when instances don't come up we don't get any logs.

**Comment by dustymabe** [comment]
In order to pin NetworkManager I also need to denylist the `ext.config.networking.default-network-behavior-change` test because we [already updated it](https://github.com/coreos/fedora-coreos-config/pull/2058) for the new NM version.

**Comment by dustymabe** [comment]
For the record - we reverted NetworkManager in https://github.com/coreos/fedora-coreos-config/pull/2075 and the azure tests are now passing again.

**Comment by dustymabe** [comment]
This appears to also affect `NetworkManager-1.40.2-1.fc37` in Fedora 37.

**Comment by dustymabe** [comment]
Pin in `next-devel`: https://github.com/coreos/fedora-coreos-config/pull/2084

**Comment by dustymabe** [comment]
This appears to be caused by [1656d82](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/1656d82).  In some cases it causes [NM in the initrd to retry forever](https://bugzilla.redhat.com/show_bug.cgi?id=2039734#c5) (it times out after one hour) and was reverted upstream in [aec7ae8](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/commit/aec7ae8).  

There have been tags created upstream with this patch included (`1.41.8-dev`, `1.40.10`) but they haven't made it into any Fedora RPM builds yet. I confirmed with `NetworkManager-1.40.11-31053.copr.285f692ac5.fc37`  from [copr](https://copr.fedorainfracloud.org/coprs/networkmanager/NetworkManager-1.40/).  that it appears to fix the problem.

The reason I think this hit us on Azure and not on other platforms is that we currently rely on falling back there for boot. See https://github.com/coreos/afterburn/issues/652

**Comment by dustymabe** [comment]
This is now taken care of. This problem never reached any production streams.
